whylog/config/rule.py

```diff
@@ -32,6 +33,29 @@ def get_causes_parsers(self):
     def get_effect_name(self):
         return self._effect.name
 
+    def _gather_causes_frequency_information(self):
+        causes_with_frequency_info = [(self._causes[0].name, 0)]
+        for cause in self._causes:
+            if cause.name == causes_with_frequency_info[-1][0]:
+                causes_with_frequency_info[-1] = (cause.name, causes_with_frequency_info[-1][1] + 1)
+            else:
+                causes_with_frequency_info.append((cause.name, 1))
+        return causes_with_frequency_info
+
+    def constraints_check(self, clues, effect_clues_dict):
+        """
+        check if given clues satisfy rule
+        basing on its causes, effect and constraints.
+        returns list of InvestigationResult objects
+        """
+        causes_with_freq_info = self._gather_causes_frequency_information()
+        clues_lists = [
+            (clues[cause_with_info[0]], cause_with_info[1])
+            for cause_with_info in causes_with_freq_info if cause_with_info[0] in clues
```

Comment:
In this way you lost clues_groups mapping. for example:
causes [a, b, a] you will create combinations something like [clue from a, clue from a, clue from b]'